http/client.go

```diff
@@ -290,7 +290,9 @@ func (c *Client) sendRateLimitedRequest(
 		if err != nil {
 			return nil, errors.Newf(err, "failed executing the request %s", URL)
 		}
-		if resp.StatusCode != http.StatusRequestEntityTooLarge || resp.Header.Get("Retry-After") == "" {
+		if (resp.StatusCode != http.StatusRequestEntityTooLarge &&
+			resp.StatusCode != http.StatusServiceUnavailable) ||
```

Comment:
Per the bug 1682827, in some cases, openstack is returning an error code 503 with the message "The maximum request receiving rate is exceeded."  So far 503 is undocumented.  

We currently check for 413.  I'll add 403 also.